core/src/frame_lifecycle.rs

```diff
@@ -112,9 +116,10 @@ pub fn run_all_phases_avm2(context: &mut UpdateContext<'_, '_>) {
 /// This is a no-op on AVM1, which has it's own catch-up logic.
 pub fn catchup_display_object_to_frame<'gc>(
     context: &mut UpdateContext<'_, 'gc>,
+    is_content_as3: bool,
     dobj: DisplayObject<'gc>,
 ) {
-    if !context.is_action_script_3() {
+    if !is_content_as3 {
```

Comment:
Can this just be `dobj.movie().is_action_script_3()`, like you do elsewhere in this PR?

Reply:
I got borrow errors when doing that, so I left it as this.